packages/analytics-browser/src/plugins/diagnostic.ts

```diff
@@ -0,0 +1,8 @@
+import { Diagnostic as CoreDiagnostic } from '@amplitude/analytics-core';
+
+export class Diagnostic extends CoreDiagnostic {
+  async flush(): Promise<void> {
+    await fetch(this.serverUrl, this.requestPayloadBuilder(this.queue));
```

Comment:
the pattern is we pass a `transportProvider` instead of creating subclass

Reply:
`transportProvider` has `send(serverUrl: string, payload: Payload)` where Payload is designed for normal events and is not compatible with diagnostic events.

Reply:
After team discussion with Ondrej and Justin, we decided that in this version, we just use fetch method instead of re-using transportProvider